dlf/common/class.tx_dlf_solr.php

```diff
@@ -136,7 +136,7 @@ public static function escapeQuery($query) {
 	public static function escapeQueryKeepField($query, $pid) {
 
 		// Is there a field query?
-		if (preg_match('/^[[:alnum:]]+_[tu][su]i:\(.*\)$/', $query)) {
+		if (preg_match('/^[[:alnum:]]+_[tu][su]i:(.*)$/', $query)) {
 
```

Comment:
AFAIK the Solr query syntax is `[field name]:([search term])`, so the parentheses are part of the query syntax, not a regex subexpression. Therefore they should be treated as literal characters and must be masked. Or am I wrong?

Reply:
I think the question is if the parentheses are optional or not. The syntax is first and foremost [field name]:[search term] but you can optionally group the search term if you want to search for more than one word or phrase in the same field e.g [field name]:([term1] [term2]).

We should change the regex to make the parentheses optional i.e. `\(?.*\)?`. What do you think, @sebastian-meyer?

That might be the case why it didn’t work for @albig. The query should work without the grouping but didn’t because they were expected.